packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -4,40 +4,37 @@
 
 import '../base/common.dart';
 import '../base/context.dart';
-import '../build_info.dart';
 import '../build_system/build_system.dart';
-import '../convert.dart';
+import '../build_system/targets/assets.dart';
+import '../build_system/targets/dart.dart';
+import '../build_system/targets/ios.dart';
+import '../build_system/targets/linux.dart';
+import '../build_system/targets/macos.dart';
+import '../build_system/targets/windows.dart';
 import '../globals.dart';
 import '../project.dart';
 import '../runner/flutter_command.dart';
 
 /// The [BuildSystem] instance.
 BuildSystem get buildSystem => context.get<BuildSystem>();
 
+/// All currently implemented targets.
+const List<Target> _kDefaultTargets = <Target>[
+  UnpackMacOS(),
+  UnpackLinux(),
+  UnpackWindows(),
+  CopyAssets(),
+  KernelSnapshot(),
+  AotElfProfile(),
+  AotElfRelease(),
+  AotAssemblyProfile(),
+  AotAssemblyRelease(),
+];
+
 /// Assemble provides a low level API to interact with the flutter tool build
 /// system.
 class AssembleCommand extends FlutterCommand {
   AssembleCommand() {
-    addSubcommand(AssembleRun());
-    addSubcommand(AssembleDescribe());
```

Comment:
Were the describe, etc. subcommands not useful? Have you considered exposing them under flags to `assemble` for e.g. dependency debugging?

Reply:
For the iOS case, I need to generate/reference a file for the xcodeproj to read so having this as a command doesn't help. Gradle might be able to use the commands, but probably not as JSON - something like `flutter assemble --list-outputs foo` which lists the outputfiles on stdout